### PR TITLE
Update sbt version to 1.7.1

### PR DESCRIPTION
**Pull request by djx314**
sbt 1.7.1 need to provide a crossScalaVersion before `++scalaVersion`, just run a command when start.

### CODE REVIEW

**Comment by Atry** [comment]
Could you fix the test failures in GitHub Actions, please?

**Comment by djx314** [comment]
@Atry Fixed the problem with a slightly tougher approach.  
[github action page](https://github.com/xarcher/dsl-domains-cats/actions/runs/3165643577)

File: .github/workflows/scala.yml [hunk 1]
```diff
@@ -51,7 +51,7 @@ jobs:
           ${{ runner.os }}-${{matrix.scala}}-${{ hashFiles('**/*.sbt') }}
           ${{ runner.os }}-${{matrix.scala}}-
     - name: Run tests
-      run: sbt ${{matrix.sbt-args}} ++${{ matrix.scala }} test
+      run: sbt ${{matrix.sbt-args}} "set crossScalaVersions := Seq(\"${{ matrix.scala }}\")" "set \`domains-catsJVM\` / crossScalaVersions := Seq(\"${{ matrix.scala }}\")" "set \`domains-catsJS\` / crossScalaVersions := Seq(\"${{ matrix.scala }}\")" "++${{ matrix.scala }} test"
```
Comment (Atry) [inline .github/workflows/scala.yml:54]:
```suggestion
      run: sbt ${{matrix.sbt-args}} ++${{ matrix.scala }} test
```

Comment (Atry) [reply .github/workflows/scala.yml:54]: `crossScalaVersions` seems unnecessary

**Review by Atry — CHANGES_REQUESTED** [review]

File: .github/workflows/scala.yml (hunk 1 above)
Comment (djx314) [reply .github/workflows/scala.yml:54]: It's the result of a compromise.
See 
https://github.com/xarcher/dsl-domains-cats/pull/2
https://github.com/xarcher/dsl-domains-cats/pull/3
It seems that sbt now only can `++` a existing crossScalaVersion.

Comment (Atry) [reply .github/workflows/scala.yml:54]: Then just add an `!`, please

Comment (djx314) [reply .github/workflows/scala.yml]: Thanks for the suggestion. Find the answer in
https://github.com/sbt/sbt/issues/3698#issuecomment-340876280
Already accepted the change. See
https://github.com/xarcher/dsl-domains-cats/actions/runs/3166811897

Comment (djx314) [reply .github/workflows/scala.yml]: Sorry for my ignorance of the pull request function.😥

**Comment by Atry** [comment]
`crossScalaVersion` should be set in the `GithubActionsScalaVersions` sbt plugin

**Comment by Atry** [comment]
I am adding support for `include` in https://github.com/ThoughtWorksInc/sbt-best-practice/pull/267

**Comment by djx314** [comment]
> I am adding support for `include` in [ThoughtWorksInc/sbt-best-practice#267](https://github.com/ThoughtWorksInc/sbt-best-practice/pull/267)

If `sbt-best-practice` can cut a release. I think this pr can merge after when the force is removed.

**Comment by Atry** [comment]
https://github.com/xarcher/dsl-domains-cats/pull/4

**Comment by djx314** [comment]
> I am adding support for `include` in [ThoughtWorksInc/sbt-best-practice#267](https://github.com/ThoughtWorksInc/sbt-best-practice/pull/267)

If `sbt-best-practice` can cut a release. I think this pr can merge after when the force is removed.

> [xarcher#4](https://github.com/xarcher/dsl-domains-cats/pull/4)

👌, Just wait before the plugin can be index.

**Comment by djx314** [comment]
> [xarcher#4](https://github.com/xarcher/dsl-domains-cats/pull/4)

Hmmm, it seems that not solve yet.

**Comment by djx314** [comment]
@Atry Build successed.

**Pull request merged**